lib/widgets/image_page.dart

```diff
@@ -120,7 +119,54 @@ class ImagePage extends StatelessWidget {
                           color: Colors.blue,
                         ),
                         onPressed: () {
-                          showUnderConstruction(context);
+                          showGeneralDialog(
```

Comment:
Can this be a a separate widget please so that the code here in here is not so difficult to read. Thanks.